pkg/apis/extensions/v1beta1/types.go

```diff
@@ -1036,3 +1037,67 @@ type PodSecurityPolicyList struct {
 	// Items is a list of schema objects.
 	Items []PodSecurityPolicy `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// Template contains the inputs needed to produce a Config.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   TemplateSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status TemplateStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
+	// Template to Config transformation.
+	Parameters []Parameter `json:"parameters,omitempty" protobuf:"bytes,2,rep,name=parameters"`
+
+	// Required: A list of resources to create
+	Objects []runtime.RawExtension `json:"objects,omitempty" protobuf:"bytes,3,rep,name=objects"`
+
+	// Optional: ObjectLabels is a set of labels that are applied to every
+	// object during the Template to Config transformation
+	ObjectLabels map[string]string `json:"objectLabels,omitempty" protobuf:"bytes,4,rep,name=objectLabels"`
+}
+
+type TemplateStatus struct {
+	State string `json:"state,omitempty" protobuf:"bytes,1,opt,name=state"`
+}
+
+// TemplateList is a list of Template objects.
+type TemplateList struct {
+	unversioned.TypeMeta `json:",inline"`
+	unversioned.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Items                []Template `json:"items,omitempty" protobuf:"bytes,2,rep,name=items"`
+}
+
+// Parameter defines a name/value variable that is to be processed during
+// the Template to Config transformation.
+type Parameter struct {
+	// Required: Parameter name must be set and it can be referenced in Template
+	// Items using ${PARAMETER_NAME}
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// Optional: The name that will show in UI instead of parameter 'Name'
+	DisplayName string `json:"displayName,omitempty" protobuf:"bytes,2,opt,name=displayName"`
+
+	// Optional: Parameter can have description
+	Description string `json:"description,omitempty" protobuf:"bytes,3,opt,name=description"`
+
+	// Optional: Value holds the Parameter data. If specified, the generator
+	// will be ignored. The value replaces all occurrences of the Parameter
+	// ${Name} expression during the Template to Config transformation.
+	Value string `json:"value,omitempty" protobuf:"bytes,4,opt,name=value"`
```

Comment:
it's a default value in case the user doesn't provide a value when instantiating the template.  for example:
https://github.com/openshift/origin/blob/master/examples/quickstarts/nodejs-mongodb.json#L376-L381